packages/gatsby-plugin-google-analytics/src/gatsby-ssr.js

```diff
@@ -83,7 +83,9 @@ export const onRenderBody = (
   }) {
     (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
     (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
-    m=s.getElementsByTagName(o)[0];a.async=1;a.defer=1;a.src=g;m.parentNode.insertBefore(a,m)
+    m=s.getElementsByTagName(o)[0];a.async=1;${
+      pluginOptions.defer ? `a.defer=1;` : ``
```

Comment:
I suspect that having both `defer` and `async` is a gray area and browsers may treat those in a different way. Here is a quote from spec:

> There are three possible modes that can be selected using these attributes. If the async attribute is present, then the script will be executed asynchronously, as soon as it is available. If the async attribute is not present but the defer attribute is present, then the script is executed when the page has finished parsing. 
> 
>The defer attribute may be specified even if the async attribute is specified, to cause legacy Web browsers that only support defer (and not async) to fall back to the defer behavior instead of the synchronous blocking behavior that is the default.

So I think we should prefer one over another but not both:

```suggestion
    m=s.getElementsByTagName(o)[0];${
      pluginOptions.defer ? `a.defer=1;` : `a.async=1;`
```

Reply:
![Async-defer](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/loading-third-party-javascript/images/image_13.png)

Ok I could do that